File: Java/LeetCode/The Two Sneaky Numbers of Digitville.java
// https://leetcode.com/problems/the-two-sneaky-numbers-of-digitville/

class Solution {

    public int[] getSneakyNumbers(int[] nums) {
        Set<Integer> uniqueNums = new HashSet<>();
        List<Integer> sneakyNums = new ArrayList<>();
        
        for (int num : nums) {
            if (uniqueNums.contains(num)) {
                sneakyNums.add(num);
            } else {
                uniqueNums.add(num);
            }
        }

        return new int[] { sneakyNums.get(0), sneakyNums.get(1) };
    }

}

File: Kotlin/LeetCode/The Two Sneaky Numbers of Digitville.kt
// https://leetcode.com/problems/the-two-sneaky-numbers-of-digitville/

class Solution {

    fun getSneakyNumbers(nums: IntArray): IntArray {
        val uniqueNums = mutableSetOf<Int>();
        val sneakyNums = mutableListOf<Int>();

        nums.forEach {
            if (uniqueNums.contains(it)) {
                sneakyNums.add(it);
            } else {
                uniqueNums.add(it);
            }
        }

        return intArrayOf(sneakyNums[0], sneakyNums[1]);
    }

}
